services/src/main/java/org/keycloak/theme/DefaultThemeManager.java

```diff
@@ -276,6 +277,32 @@ private Map<Locale, Properties> getMessagesByLocale(String baseBundlename, Local
             }
         }
 
+        protected void addlocaleTranslations(Locale locale, Properties m) throws IOException {
+            for (String l : getProperties().getProperty("locales", "").split(",")) {
+                l = l.trim();
+                String key = "locale_" + l;
+                String label = m.getProperty(key);
+                if (label == null) {
```

Comment:
Alternatively this could be inversed to reduce nesting, but more of a code-style thing:

```java
if (label != null) {
  continue;
}
```

Reply:
part of next commit